### PR TITLE
Pass explicit node type to LiteralInterpreter

### PR DESCRIPTION
**Pull request by sopel39**
Follow up of https://github.com/trinodb/trino/pull/11918

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/cost/ScalarStatsCalculator.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ protected SymbolStatsEstimate visitLiteral(Literal node, Void context)
         {
             ExpressionAnalyzer analyzer = createConstantAnalyzer(plannerContext, new AllowAllAccessControl(), session, ImmutableMap.of(), WarningCollector.NOOP);
             Type type = analyzer.analyze(node, Scope.create());
-            Object value = literalInterpreter.evaluate(analyzer.getExpressionTypes(), node);
+            Object value = literalInterpreter.evaluate(node, analyzer.getExpressionTypes().get(NodeRef.of(node)));
```
Comment (kasiafi) [inline core/trino-main/src/main/java/io/trino/cost/ScalarStatsCalculator.java:121]:
```suggestion
            Object value = literalInterpreter.evaluate(node, type);
```

File: core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java [hunk 2]
```diff
@@ -389,7 +389,7 @@ protected Object visitSymbolReference(SymbolReference node, Object context)
         @Override
         protected Object visitLiteral(Literal node, Object context)
         {
-            return literalInterpreter.evaluate(expressionTypes, node);
+            return literalInterpreter.evaluate(node, expressionTypes.get(NodeRef.of(node)));
```
Comment (kasiafi) [inline core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java:392]:
```suggestion
            return literalInterpreter.evaluate(node, type(node));
```

**Review by kasiafi — APPROVED** [review]

**Review by findepi — COMMENTED** [review]
LGTM % @kasiafi 's comments

**Review by findepi — APPROVED** [review]
nice, thanks

**Pull request merged**